### PR TITLE
Adapt MapMakerObs to return a MapDataset

### PR DESCRIPTION
**Pull request by AtreyeeS**
This PR adapts `MapMakerObs` to return a `MapDataset` instead of a dictionary.
This needs some adaption in `MapMaker` and `MapMakerRing` as well. Since these classes will move away in near future, I have put in some temporary ugly lines.

### CODE REVIEW

File: gammapy/cube/make.py [hunk 1]
```diff
@@ -268,9 +275,30 @@ def run(self, selection=None):
         for name in selection:
             getattr(self, "_make_" + name)()
 
+        background_model = BackgroundModel(self.maps["background"])
+
+        if "psf" in self.maps:
```
Comment (adonath) [inline gammapy/cube/make.py]: As `self.maps` is a dictionary the `if` and `else` blocks could be removed and replaced by `psf = self.maps.get("psf")`

File: gammapy/cube/make.py [hunk 2]
```diff
@@ -268,9 +275,30 @@ def run(self, selection=None):
...
+            edisp = self.maps["edisp"]
+        else:
+            edisp = None
+
+        dataset = MapDataset(
+            counts=self.maps["counts"],
+            exposure=self.maps["exposure"],
+            background_model=background_model,
+            psf=psf,
+            edisp=edisp,
```
Comment (adonath) [inline gammapy/cube/make.py]: As there is the access to the `self.observation` objects, you could also copy over the `GTI` table here and maybe set the name of the dataset to `"obs-{}".format(observation.obs_id)` or similar.

Comment (AtreyeeS) [reply gammapy/cube/make.py]: Thanks @adonath ! Addressed the changes...

**Comment by adonath** [comment]
Thanks a lot @AtreyeeS! Once we have the `MapDataset.stack()` method implemented we should consider adapting our tutorials to directly use the `MapMakerObs` and maybe also rename it then...

**Pull request merged**